tests/NCronJob.Tests/NotificationHandlerTests.cs

```diff
@@ -87,13 +73,24 @@ public async Task HandlerThatThrowsExceptionInAsyncPartShouldNotInfluenceOtherHa
                 .AddJob<ExceptionJob>(p => p.WithCronExpression(Cron.AtEveryMinute))
                 .AddNotificationHandler<HandlerThatThrowsInAsyncPartException>()
         );
-        var provider = CreateServiceProvider();
 
-        await provider.GetRequiredService<IHostedService>().StartAsync(CancellationToken);
+        await AssertSimpleJobWasProcessedAndNotified();
+    }
+
+    private async Task AssertSimpleJobWasProcessedAndNotified()
+    {
+        (IDisposable subscription, IList<ExecutionProgress> events) = RegisterAnExecutionProgressSubscriber(ServiceProvider);
+
+        await ServiceProvider.GetRequiredService<IHostedService>().StartAsync(CancellationToken);
+
+        Guid orchestrationId = events.First().CorrelationId;
+
+        await WaitForOrchestrationCompletion(events, orchestrationId);
+
+        subscription.Dispose();
 
-        FakeTimer.Advance(TimeSpan.FromMinutes(1));
-        var message = await CommunicationChannel.Reader.ReadAsync(CancellationToken);
-        message.ShouldBe("Foo");
+        Assert.Single(Storage.Entries);
```

Comment:
Also to stick to one pattern: Can we make this `Shouldly` style?

I am fare if we remove shouldly entirely, but one I'd prefer if we stick to "one theme"

Reply:
Shouldy it is!